netCDF4/_netCDF4.pyx

```diff
@@ -1628,10 +1628,10 @@ cdef _get_vars(group):
         free(varids) # free pointer holding variable ids.
     return variables
 
-cdef _ensure_nc_success(ierr, err_cls=RuntimeError):
+cdef _ensure_nc_success(ierr, *args, err_cls=RuntimeError):
     # print netcdf error message, raise error.
     if ierr != NC_NOERR:
-        raise err_cls((<char *>nc_strerror(ierr)).decode('ascii'))
```

Comment:
DIffernet error classes use different constructors — it’s not always valid to pass an error number as the first argument. I think this function needs a path specifically for `EnvironmentError`.